src/extractor/scripting_environment.cpp

```diff
@@ -51,122 +52,137 @@ int luaErrorCallback(lua_State *state)
 }
 }
 
+ScriptingEnvironment::Context::Context() : state(luaL_newstate()) {}
+
+ScriptingEnvironment::Context::~Context() { lua_close(state); }
```

Comment:
I feel like we should abstract away the Lua state resource ownership and then hold onto an object by-value via the Rule Of Zero (which means we don't need ctor/dtor/{move,copy}{assign,construct} any longer).

What I mean is e.g. doing https://gist.github.com/daniel-j-h/7ca2bf73047c788f3d9b#file-luaraii-cc-L14-L24
and then holding on to a state by value and composing it with other member attributes.